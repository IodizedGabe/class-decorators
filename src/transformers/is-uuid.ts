import { composePropertyDecorators, IsUUIDOptions } from '@hippo-oss/dto-decorators';

import { createBasePropertyDecorators } from './base';

export function IsUUID(options: IsUUIDOptions): PropertyDecorator {
    return composePropertyDecorators([
        ...createBasePropertyDecorators(options),
    ]);
}
