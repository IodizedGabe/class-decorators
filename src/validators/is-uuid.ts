import { composePropertyDecorators, IsUUIDOptions } from '@hippo-oss/dto-decorators';
import { IsUUID as IsUUIDDecorator } from 'class-validator';

import { createBasePropertyDecorators } from './base';

export function IsUUID(options: IsUUIDOptions = {}): PropertyDecorator {
    return composePropertyDecorators([
        ...createBasePropertyDecorators(options),

        // validate data as a string
        IsUUIDDecorator(options.version, { each: !!options.isArray }),
    ]);
}
