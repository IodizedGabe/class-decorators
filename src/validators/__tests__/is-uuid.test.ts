import { validateSync } from 'class-validator';

import { IsUUID } from '../is-uuid';

describe('IsUUID', () => {
    it('allows optional uuid to be omitted', () => {
        class Fixture {
            @IsUUID({
                optional: true,
            })
            property!: string;
        }

        const obj = new Fixture();
        const errors = validateSync(obj);
        expect(errors).toHaveLength(0);
    });
    it('allows nullable uuid to be omitted', () => {
        class Fixture {
            @IsUUID({
                nullable: true,
            })
            property!: string;
        }

        const obj = new Fixture();
        const errors = validateSync(obj);
        expect(errors).toHaveLength(0);
    });
    it('does not allow required uuid to be omitted', () => {
        class Fixture {
            @IsUUID({})
            property!: string;
        }

        const obj = new Fixture();
        const errors = validateSync(obj);
        expect(errors).toHaveLength(1);
        expect(errors).toMatchSnapshot();
    });
    it('does not allow a non-uuid of the wrong type', () => {
        class Fixture {
            @IsUUID({})
            property!: string;
        }

        const obj = new Fixture();
        obj.property = 'not-a-uuid';
        const errors = validateSync(obj);
        expect(errors).toHaveLength(1);
        expect(errors).toMatchSnapshot();
    });
    it('does not allow a required uuid of the wrong version', () => {
        class Fixture {
            @IsUUID({
                version: '4',
            })
            property!: string;
        }

        const obj = new Fixture();
        obj.property = 'c50318b5-a0d1-11ec-b448-8c85905fd9db';
        const errors = validateSync(obj);
        expect(errors).toHaveLength(1);
        expect(errors).toMatchSnapshot();
    });
    it('allows a required uuid of the correct version', () => {
        class Fixture {
            @IsUUID({
                version: '4',
            })
            property!: string;
        }

        const obj = new Fixture();
        obj.property = '09ea1206-495e-444b-afb1-9e35a1eb0545';
        const errors = validateSync(obj);
        expect(errors).toHaveLength(0);
    });
});
