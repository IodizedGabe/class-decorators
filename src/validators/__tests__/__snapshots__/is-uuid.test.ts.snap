// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IsUUID does not allow a non-uuid of the wrong type 1`] = `
Array [
  ValidationError {
    "children": Array [],
    "constraints": Object {
      "isUuid": "property must be a UUID",
    },
    "property": "property",
    "target": Fixture {
      "property": "not-a-uuid",
    },
    "value": "not-a-uuid",
  },
]
`;

exports[`IsUUID does not allow a required uuid of the wrong version 1`] = `
Array [
  ValidationError {
    "children": Array [],
    "constraints": Object {
      "isUuid": "property must be a UUID",
    },
    "property": "property",
    "target": Fixture {
      "property": "c50318b5-a0d1-11ec-b448-8c85905fd9db",
    },
    "value": "c50318b5-a0d1-11ec-b448-8c85905fd9db",
  },
]
`;

exports[`IsUUID does not allow required uuid to be omitted 1`] = `
Array [
  ValidationError {
    "children": Array [],
    "constraints": Object {
      "isDefined": "property should not be null or undefined",
      "isUuid": "property must be a UUID",
    },
    "property": "property",
    "target": Fixture {},
    "value": undefined,
  },
]
`;
